lib/facility_factory.rb

```diff
@@ -0,0 +1,43 @@
+class FacilityFactory
+
+    def initialize
+    end
+
+    def create_facilities(facilities)
+        format_data(facilities)
+        facilities.map do |facility|
+            Facility.new(facility)
+        end
+    end
+
+    def format_data(facilities)
+        if facilities.first.has_key?(:dmv_office)
```

Comment:
Creating a different helper method for each state is an excellent way to break up this method. Well done!